rest/config.go

```diff
@@ -145,6 +146,10 @@ type DbConfig struct {
 	SendWWWAuthenticateHeader    *bool                          `json:"send_www_authenticate_header,omitempty"` // If false, disables setting of 'WWW-Authenticate' header in 401 responses
 }
 
+type DeprecatedOptions struct {
+	Shadow *ShadowConfig `json:"shadow,omitempty"` // External bucket to shadow
+}
+
```

Comment:
What's the advantage of DeprecatedOptions over the already existing Unsupported?

Reply:
The main motivation was to work around circular packages issues.  `db.UnsupportedOptions` is in the `db` package, and `ShadowConfig` is in the `rest` package.  Making `db` reference `rest` looks like it would create a circular dependency, since `rest` already references `db`.

However I think the naming is also clearer since it distinguishes between "experimental" features that are upcoming and "deprecated" features that are being removed.

Reply:
That's reasonable - thanks for the details.